typescript/alert-center/src/operations.ts

```diff
@@ -0,0 +1,106 @@
+  });
+
+const timeToPackAlert = 30;
+
+export interface AlertEmployeeInfo
+{
+  employee: Employee;
+  alert: AlertEmployee[];
+  expirationSecs: number | null;
+  newAssignment: boolean;
+}
+
+@Kafka(kafkaConfig)
+export class Respondment {
+  @Workflow()
+  @KafkaConsume(respondTopic)
+  static async inboundAlertWorkflow(ctxt: WorkflowContext, topic: string, _partition: number, message: KafkaMessage) {
+    if (topic !== respondTopic) return; // Error
+
+    const payload = JSON.parse(message.value!.toString()) as {
+      alerts: AlertWithMessage[],
+    };
+
+    ctxt.logger.info(`Received alert: ${JSON.stringify(payload)}`);
+
+    for (const detail of payload.alerts) {
+      if (detail.alert_status !== AlertStatus.ACTIVE) continue;
+      await ctxt.invoke(RespondUtilities).addAlert(detail);
+    }
+
+    return Promise.resolve();
+  }
+
+  @Workflow()
+  static async userAssignmentWorkflow(ctxt: WorkflowContext, name: string, @ArgOptional more_time: boolean | undefined) {
+    let ctime = await ctxt.invoke(CurrentTimeCommunicator).getCurrentTime();
+    const expiration = ctime + timeToPackAlert*1000;
+    const userRec = await ctxt.invoke(RespondUtilities).getUserAssignment(name, new Date(expiration), more_time);
+    const expirationSecs = userRec.employee.expiration ? (userRec.employee.expiration!.getTime()-ctime) / 1000 : null;
+    await ctxt.setEvent<AlertEmployeeInfo>('rec', {...userRec, expirationSecs});
+      
+    if (userRec.newAssignment) {
+      ctxt.logger.info(`Start watch workflow for ${name}`);
+      // Keep a watch over the expiration...
+      let expirationMS = userRec.employee.expiration ? userRec.employee.expiration.getTime() : 0;
+      while (expirationMS > ctime) {
+        ctxt.logger.debug(`Sleeping ${expirationMS-ctime}`);
+        await ctxt.sleepms(expirationMS - ctime);
+        const curDate = await ctxt.invoke(CurrentTimeCommunicator).getCurrentDate();
+        ctime = curDate.getTime();
+        const nextTime = await ctxt.invoke(RespondUtilities).checkForExpiredAssignment(name, curDate);
+        if (!nextTime) {
+          ctxt.logger.info(`Assignment for ${name} ended; no longer watching.`);
+          break;
+        }
+        expirationMS = nextTime.getTime();
+        ctxt.logger.info(`Going around again: ${expirationMS} / ${ctime}`);
+      }
+    }
+  }
+
+  @PostApi('/crash_application')
+  static async crashApplication(_ctxt: HandlerContext) {
+    // For testing and demo purposes :)
+    process.exit(1);
+    return Promise.resolve();
+  }
+
+  @PostApi('/do_send')
+  @Workflow()
+  static async sendAlert(ctxt: WorkflowContext, message: string) {
+      const max_id = await ctxt.invoke(RespondUtilities).getMaxId()  
+      await ctxt.invoke(producerConfig).sendMessage(
+      {
+        value: JSON.stringify({
+          alerts: [
+            { 
+              alert_id: max_id+1,
+              alert_status: AlertStatus.ACTIVE,
+              last_update_time: "2024-01-01",
```

Comment:
We can use the actual time, `Date.now().toISOString()`. Why do we hardcode the time here?

Reply:
actually Date.now() is not allowed per our linter :D 

But I simply removed it from the structure, it's not used